### PR TITLE
Release instructions

### PR DESCRIPTION
**Pull request by azatsarynnyy**
### What does this PR do?
Provides `RELEASE.md` file with the release instructions.

### What issues does this PR fix or reference?
https://github.com/eclipse/che/issues/15506

### CODE REVIEW

File: RELEASE.md [hunk 1]
```diff
@@ -0,0 +1,107 @@
+# Che Theia release process
+
+#### Make the release changes
+
+1. Create a branch for the release, e.g. `7.7.0`
```
Comment (benoitf) [inline RELEASE.md]: just a notice, but in other repos, branches are named `7.7.x`
https://github.com/eclipse/che/tree/7.7.x

`7.7.0` is often used as a `tag name`

Comment (azatsarynnyy) [reply RELEASE.md]: yeah, I saw that
I've just decided to keep it as I received from the former IDE2 team so as not to accidentally break something

Comment (benoitf) [reply RELEASE.md]: sure, just maybe try for the upcoming releases to match other repositories

Comment (azatsarynnyy) [reply RELEASE.md]: I'll try

Comment (azatsarynnyy) [reply RELEASE.md]: proposed a PR https://github.com/eclipse/che-theia/pull/630

**Pull request merged**